File: Data Structures & Algorithms/jump-game/submission-1.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    canJump(nums: number[]): boolean {
        function check(i: number): boolean {
            let num = nums[i];
            console.log(num);
            if (i >= nums.length - 1) return true;
            if (num === 0) return false;
            for (let j = i+1; j <= i+num; j++) {
                if (check(j)) {
                    return true;
                }
            }
            return false;
        }
        return check(0);
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = {}
        for num in nums:
            freq[num] = freq.get(num, 0) + 1

        print(freq)
        bucket = [[] for _ in range(len(nums)+1)]

        for key, value in freq.items():
            bucket[value].append(int(key))

        results = []
        for i in range(len(nums) , -1, -1):
            if bucket[i] is not None:
                results.extend(bucket[i])
            if len(results) == k:
                return results
        return results
